src/main/java/dev/Roach/DealsFetcher.java

```diff
@@ -0,0 +1,79 @@
+package dev.Roach;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+public class DealsFetcher {
+
+    private final HttpClient client;
+
+    public DealsFetcher(HttpClient client) {
+        this.client = client;
+    }
+
+    public List<String> getAllDeals() {
+
+        List<String> allPages = new ArrayList<>();
+
+        HttpRequest initialRequest = HttpRequest.newBuilder()
+                .uri(URI.create("https://www.cheapshark.com/api/1.0/deals?pageNumber=0"))
+                .GET()
+                .build();
+
+        try (FileWriter fw = new FileWriter("listOfAllDeals.txt")) {
+
+            HttpResponse<String> initialResponse = client.send(initialRequest, HttpResponse.BodyHandlers.ofString());
+            int totalPages = Integer.parseInt(initialResponse.headers().firstValue("X-Total-Page-Count").orElse("0"));
```

Comment:
is orElse required here? Is it possible for the header to not include page number at all?

Reply:
Yes, you're right; I didn't think about that. In the worst case, the value will be 0, just like in my 'orElse' condition, right? I will remove this.